spec/lexers/abap_spec.rb

```diff
@@ -2,7 +2,8 @@
 # frozen_string_literal: true
 
 describe Rouge::Lexers::ABAP do
-  let(:subject) { Rouge::Lexers::ABAP.new }
+  let(:described_class) { Rouge::Lexers::ABAP }
```

Comment:
Is this meant to be in the PR?

Reply:
Yes. `Rouge::Lexers::ABAP` doesn't implement the `:detect?` singleton_method. So it is being used to test for false positive response from the new singleton_method `:detectable?`

Reply:
That feels a bit brittle. I think the appropriate place to put the test would be in `spec/lexer_spec.rb`. Something like:

```rb
it 'is detectable' do
  class DetectableLexer < Rouge::RegexLexer
    def self.detect?
      return true
    end
  end

  assert DetectableLexer.detectable?
end

it 'is undetectable' do
  class UndetectableLexer < Rouge::RegexLexer
  end

  refute UndetectableLexer.detectable?
end
```